### DISCUSSION TITLE
fcitx5停止时，导致Alacritty界面卡住

### DISCUSSION THREAD

**Discussion by bczhc** [fcitx/fcitx5#1096] [General]
Alacritty那边的issue： alacritty/alacritty#7456

我是感觉这*应该*不是fcitx5的问题，不过时隔半年多，依然没解决，在这边提一下吧。不知fcitx5开发组有没有啥头绪，以及，有没有其他遇到同样问题的。任何想法都很感谢！

**Comment by wengxt** [comment]
一般来说，谁 freeze 就是哪一方（作为整体）的问题。

但不代表他们就能修复这个问题。alacritty 使用 winit 用于显示，winit 用 libX11 作为输入法的实现的库。而 libX11 本身的 xim 实现就有遇到各种情况就卡死在 libX11 里面的问题，这不是 alacritty 本身能控制的

**Reply by bczhc** [reply]
我发现了一些新的现象：

[a.webm](https://github.com/user-attachments/assets/c2e086a3-885f-4284-8b1a-6983af630752)

先把Rime addon去掉，只留English。把Fcitx5-configtool打开，再启动fcitx5，当我只kill fcitx5时，Alacritty不卡死，但当我同时杀，`pkill fcitx5`，或者`kill <configtool-pid>; kill <fcitx5-pid>`时，Alacritty卡死。

**Reply by bczhc** [reply]
以及，用同样有composition步骤的`fcitx5-kkc`日文输入插件，好像……也没触发？……我试的其他几个终端（xfce4-terminal，Kitty等）都没问题，但Alacritty貌似不打算动x11后端了……

[b.webm](https://github.com/user-attachments/assets/db17e8b4-aff6-4e26-bd47-d152af2bd734)
